tools/fuzzer.cpp

```diff
@@ -0,0 +1,403 @@
+          generateVelocypack<Format>(builder, depth + 1, randomGenerator);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_BOOLEAN:
+        builder.add(Value(randomGenerator.mt64() % 2 ? true : false));
+        break;
+      case ADD_STRING: {
+        std::string key;
+        generateUtf8String(randomGenerator, key);
+        builder.add(Value(key));
+        break;
+      }
+      case ADD_NULL:
+        builder.add(Value(ValueType::Null));
+        break;
+      case ADD_UINT64:
+        builder.add(Value(randomGenerator.mt64()));
+        break;
+      case ADD_INT64: {
+        uint64_t uintValue = randomGenerator.mt64();
+        int64_t intValue;
+        memcpy(&intValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(intValue));
+        break;
+      }
+      case ADD_DOUBLE: {
+        double doubleValue;
+        do {
+          uint32_t uintValue = static_cast<uint32_t>(randomGenerator.mt64());
+          memcpy(&doubleValue, &uintValue, sizeof(uintValue));
+        } while (!std::isfinite(doubleValue));
+        builder.add(Value(doubleValue));
+        break;
+      }
+      case ADD_UTC_DATE:
+        builder.add(Value(randomGenerator.mt64(), ValueType::UTCDate));
+        break;
+      case ADD_BINARY: {
+        std::string binaries;
+        generateUtf8String(randomGenerator, binaries);
+        builder.add(ValuePair(binaries.data(), binaries.size(), ValueType::Binary));
+        break;
+      }
+      case ADD_EXTERNAL: {
+        builder.add(Value(static_cast<void const *>(&nullSlice), ValueType::External));
+        break;
+      }
+      case ADD_ILLEGAL:
+        builder.add(Value(ValueType::Illegal));
+      break;
+      case ADD_MIN_KEY:
+        builder.add(Value(ValueType::MinKey));
+      break;
+      case ADD_MAX_KEY:
+        builder.add(Value(ValueType::MaxKey));
+      break;
+      default:
+        assert(false);
+    }
+    break;
+  }
+}
+
+static bool isParamValid(char const* p, uint64_t& value) {
+  auto result = std::from_chars(p, p + strlen(p), value);
+  if (result.ec != std::errc()) {
+    std::cerr << "Error: wrong parameter type: " << p << std::endl;
+    return false;
+  }
+  return true;
+}
+
+int main(int argc, char* argv[]) {
+  VELOCYPACK_GLOBAL_EXCEPTION_TRY
+    bool isTypeAssigned = false;
+    uint32_t numIterations = 1;
+    uint32_t numThreads = 1;
+    bool isJSON = false;
+    std::random_device rd;
+    uint64_t seed = rd();
+
+    int i = 1;
+    while (i < argc) {
+      bool isFailure = false;
+      char const *p = argv[i];
+      if (isOption(p, "--help")) {
+        usage(argv);
+        return EXIT_SUCCESS;
+      } else if (isOption(p, "--vpack") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        isJSON = false;
+      } else if (isOption(p, "--json") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        isJSON = true;
+      } else if (isOption(p, "--i")) {
+        char const *p = argv[++i];
```

Comment:
Security issue: we must check if `++i >= argc`. Otherwise we read out of bounds, e.g. someone specified `fuzzer -i` (without specifying the number of iterations).